examples/features/metadata/client/main.go

```diff
@@ -71,15 +71,14 @@ func unaryCallWithMetadata(c pb.EchoClient, message string) {
 	}
 	fmt.Printf("response:\n")
 	fmt.Printf(" - %s\n", r.Message)
-
-	if t, ok := trailer["timestamp"]; ok {
-		fmt.Printf("timestamp from trailer:\n")
-		for i, e := range t {
-			fmt.Printf(" %d. %s\n", i, e)
-		}
-	} else {
+	t, ok := trailer["timestamp"]
+	if !ok {
 		log.Fatal("timestamp expected but doesn't exist in trailer")
 	}
+	fmt.Printf("timestamp from trailer:\n")
+	for i, e := range t {
+		fmt.Printf(" %d. %s\n", i, e)
+	}
```

Comment:
This refactoring doesn't increase the scope of `t, ok` immediately. However, when more more code is added at the end of the function, `t, ok` would be accessible. With the `if t, ok := trailer["timestamp"]; ok` this wouldn't happen. 

I don't think the benefit of this rule is worth the risk of causing a potential bug because of accidentally accessing a variable define higher in the function.